src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -424,28 +428,19 @@ public function next_tag( $query = null ) {
 					continue;
 				}
 
-				if ( ! parent::is_tag_closer() ) {
+				if ( ! parent::is_tag_closer() || $visit_closers ) {
```

Comment:
Should this also use `$this`?

```suggestion
if ( ! $this->is_tag_closer() || $visit_closers ) {
```

Reply:
Yes. Can you concoct a test case to fail for it? If not, don't worry - we can just fix it.